Categorization

sejkimm opened the pull request:
### **0. Categorization**

### **1. Add Route /category** 
**1.1 Add /html/CategoryList.html**
-  It renders /category

**1.2 Add func _ListCategory() in (app.go)**
- It controls /html/CategoryList.html

**1.3 Add func _getCategoryList() in (dataprocess.go)**
- Get category list form DB and returns to _ListCategory()

### **2. add Route /category/{cat}** 
**2.1 Add /html/Category.html**
-  It renders /category/{cat}. It is similar with /html/index.html but it does not cotain paging and submit code href.

**2.2 Add func _getCategoryCode() in (app.go)**
- It controls /html/Category.html

**2.3 Add func _getPostListFromCategory() in (dataprocess.go)**
- Get posts from DB and returns to _getCategoryCode(). It is similar with _getPostList() in dataprocess.go but it returns specific code from DB that matches category.

### **3. modify DB Scheme**
Check Line19~32 in dataprocess.go

### **Things to check meticulously**
- function _ListCategory in (app.go), Line 163~167. 
- /html/categorylist.html, Line 25~27.

sejkimm committed:
Add /category, /category/cat

Add Category routing

sejkimm committed:
Update and merge from origin

sejkimm committed:
Fix typo

joona0825 commented on app.go:
```diff
@@ -151,3 +154,49 @@ func _registerComment(w http.ResponseWriter, r *http.Request) {
+func _ListCategory(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	if err == nil {
```
Variable err is not declared here

hellojunha commented on app.go:
```diff
@@ -151,3 +154,49 @@ func _registerComment(w http.ResponseWriter, r *http.Request) {
+func _ListCategory(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	if err == nil {
```
Variable err is not declared

sejkimm replied:
Fixed in App.go Line 161   d3e72a5

hellojunha commented on app.go:
```diff
@@ -151,3 +154,49 @@ func _registerComment(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	page := vars["cat"]
+
+	list = getPostListFromCategory(cat)
```
No pagination needed?

sejkimm replied:
All categories have fewer than 10 posts.

hellojunha commented on app.go:
```diff
@@ -151,3 +154,49 @@ func _registerComment(w http.ResponseWriter, r *http.Request) {
+
+	if err == nil {
+		categroies := _getCategoryList() // returns string type slice 
+		t, err := template.ParseFiles(wd + "/html/categoryList.html")
```
Method for getting template file has changed. Check out the latest commit.

sejkimm replied:
Fixed

hellojunha commented on app.go:
```diff
@@ -151,3 +154,49 @@ func _registerComment(w http.ResponseWriter, r *http.Request) {
+
+	list = getPostListFromCategory(cat)
+
+	t, err := template.ParseFiles(wd + "/html/index.html")
```
Method for getting template file has changed. Check out the latest commit.

sejkimm replied:
Fixed

hellojunha commented on html/post.html:
```diff
@@ -11,7 +11,15 @@
 <form id="code_form" action="/post/register" method="post">
+    <!-- Dropdown to select Category-->
+    <select name="category">
+        <option value="LinkedList">LinkedList</option>
```
hard code?

sejkimm replied:
Fixed

hellojunha commented on dataprocess.go:
```diff
@@ -20,9 +20,17 @@ type Post struct {
 	Id int
 	Dt time.Time
 	Code string
+	Category []Category // Add relationship in DB
```
Why array?

sejkimm replied:
Fixed. Now string

hellojunha commented on dataprocess.go:
```diff
@@ -94,6 +102,67 @@ func getPostList(page int) ([]Post, int) {
+	if err == nil {
+		defer rows.Close()
+		for rows.Next() {
+			err := rows.Scan(&category)
```
Variable category is not declared

sejkimm replied:
Fixed

hellojunha commented on dataprocess.go:
```diff
@@ -94,6 +102,67 @@ func getPostList(page int) ([]Post, int) {
+}
+
+// similar as func getPostList() but Query is little different, and not paging  
+func getPostListFromCategory(string cat) [] {
```
return type?

sejkimm replied:
Fixed

joona0825 commented:
Please upload sql query I need to execute (study_category table). And you don't need too much comments in code as you wrote. I personally do not prefer writing comment because it means the code is hard to read and understand.

sejkimm committed:
Categorization (Works well)

Fixed Error

sejkimm commented:
Thanks for your review. 
I fixed several errors and It now works properly on my local environment.

SQL query for my commit is below 
`CREATE TABLE study_category ( category_id INT(11) NOT NULL AUTO_INCREMENT, category TINYTEXT NOT NULL, Week INT(11) NOT NULL, PRIMARY KEY(category_id));`

and might need one more query 
`ALTER TABLE study_post ADD COLUMN category TINYTEXT NOT NULL`

Also I removed every comment I wrote.
